### PR TITLE
Add 422 UNIDENTIFIABLE_DEVICE error in API-design-guidelines.md table

### LINKED ISSUE #249: Add 422 `UNIDENTIFIABLE_DEVICE` error in Service Exceptions table (§6.1)
**Problem description**
Very small one....

In Appendix A: info.description template for device identification from access token we can read:

>  If the `device` object is not included in the request and 
the device information cannot be derived from the 3-legged access token, the server will return a 422 `UNIDENTIFIABLE_DEVICE` error.

.... but 422 `UNIDENTIFIABLE_DEVICE` error is not in Service Exceptions table (§6.1)

**Expected action**
Add 422 `UNIDENTIFIABLE_DEVICE` error in Service Exceptions table (§6.1)

**Additional context**
@rartych as it seems to be a straightforward i can probably craft the PR? wdyt?

### PR DESCRIPTION
**Pull request by rartych**
#### What type of PR is this?

Add one of the following kinds:
* correction

#### What this PR does / why we need it:

Alignment of table in section 6.2 with Annex A.

#### Which issue(s) this PR fixes:

Fixes #249 

#### Special notes for reviewers:

To be included in RC.2

#### Changelog input

```
 422 UNIDENTIFIABLE_DEVICE error added to API-design-guidelines.md table in section 6.2

```

### CODE REVIEW

**Review by bigludo7 — APPROVED** [review]
LGTM

**Review by patrice-conil — APPROVED** [review]
LGTM

**Comment by hdamker** [comment]
@rartych #249 is mentioning "Section 6.1 and to the table in Section 6.2" but here in the PR it is only added to 6.2. By intention?

**Comment by hdamker** [comment]
In addition it would good to add it also to artifacts/CAMARA_common.yaml for consistency.

**Review by bigludo7 — APPROVED** [review]
Yes good to manage this also in the Service Exceptions Table.

File: artifacts/CAMARA_common.yaml [hunk 1]
```diff
@@ -370,6 +370,12 @@ components:
                 status: 422
                 code: DEVICE_NOT_APPLICABLE
                 message: The service is not available for the provided device.
+            GENERIC_422_UNIDENTIFIABLE_DEVICE:
+              description: Service is not available for the provided device
+              value:
+                status: 422
+                code:  UNIDENTIFIABLE_DEVICE
```
Comment (hdamker) [inline artifacts/CAMARA_common.yaml:377]: Got linter error "too many spaces after colon"

```suggestion
                code: UNIDENTIFIABLE_DEVICE

```

File: documentation/API-design-guidelines.md [hunk 2]
```diff
@@ -756,6 +756,7 @@ In the following, we elaborate on the existing client errors. In particular, we
...
 |       422        | `DEVICE_IDENTIFIERS_MISMATCH` | Provided device identifiers are not consistent.                            | Inconsistency between device identifiers not pointing to the same device                                                                                                     |
 |       422        |    `DEVICE_NOT_APPLICABLE`    | The service is not available for the provided device.                      | Service is not available for the provided device                                                                                                                             |
+|       422        |    `UNIDENTIFIABLE_DEVICE`    |  The device cannot be identified.                                          |The device identifier is not included in the request and the device information cannot be derived from the 3-legged access token                                              |
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:759]:
```suggestion
|       422        |    `UNIDENTIFIABLE_DEVICE`    | The device cannot be identified.                                          | The device identifier is not included in the request and the device information cannot be derived from the 3-legged access token                                              |
```

File: artifacts/CAMARA_common.yaml [hunk 3]
```diff
@@ -370,6 +370,13 @@ components:
                 status: 422
                 code: DEVICE_NOT_APPLICABLE
                 message: The service is not available for the provided device.
+            GENERIC_422_UNIDENTIFIABLE_DEVICE:
+              description: Service is not available for the provided device
```
Comment (PedroDiez) [inline artifacts/CAMARA_common.yaml:374]:
```suggestion
              description: The device identifier is not included in the request and the device information cannot be derived from the 3-legged access token

```

File: documentation/API-design-guidelines.md [hunk 4]
```diff
@@ -793,6 +794,9 @@ The Following table compiles the guidelines to be adopted:
 |     3      | Device identifiers mismatch                                                |       422        |  DEVICE_IDENTIFIERS_MISMATCH   | Provided device identifiers are not consistent.          |
 |     4      | Invalid access token context                                               |       403        |     INVALID_TOKEN_CONTEXT      | Device identifiers are not consistent with access token. |
 |     5      | Service not applicable to the device                                       |       422        |     DEVICE_NOT_APPLICABLE      | The service is not available for the provided device.    |
+|     6      | The device identifier is not included in the request and the device information cannot be derived from the 3-legged access token |       422        |     UNIDENTIFIABLE_DEVICE      |The device cannot be identified.|
```
Comment (PedroDiez) [inline documentation/API-design-guidelines.md:797]:
```suggestion
|     6      | The device identifier is not included in the request and the device information cannot be derived from the 3-legged access token |       422        |     UNIDENTIFIABLE_DEVICE      | The device cannot be identified. |
```

**Review by PedroDiez — COMMENTED** [review]
Some editorials for aligment.

Also 422 code can be referenced in table of section 3.2
https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#32-http-response-codes

File: artifacts/CAMARA_common.yaml [hunk 5]
```diff
@@ -370,6 +370,14 @@ components:
...
                 code: DEVICE_NOT_APPLICABLE
                 message: The service is not available for the provided device.
+            GENERIC_422_UNIDENTIFIABLE_DEVICE:
+              description: The device identifier is not included in the request and the device information cannot be derived from the 3-legged access token
+
+              value:
+                status: 422
+                code: UNIDENTIFIABLE_DEVICE
+
+                message: The device cannot be identified.
```
Comment (hdamker) [inline artifacts/CAMARA_common.yaml:380]: Delete the two empty lines
```suggestion
            GENERIC_422_UNIDENTIFIABLE_DEVICE:
              description: The device identifier is not included in the request and the device information cannot be derived from the 3-legged access token
              value:
                status: 422
                code: UNIDENTIFIABLE_DEVICE
                message: The device cannot be identified.
```

**Review by patrice-conil — DISMISSED** [review]
LGTM

**Review by hdamker — APPROVED** [review]

**Review by PedroDiez — APPROVED** [review]
LGTM

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Pull request merged**